### PR TITLE
fix(python): Fix handling of TextIOWrapper in write_csv

### PR DESCRIPTION
**Pull request by ruihe774**
Fixes "write to closed file" in `write_csv`:

```python
>>> df.write_csv(open("test.csv", "w"))
OSError: write to closed file
```

### CODE REVIEW

**Comment by ritchie46** [comment]
Thanks, can you add a test that showed the issue before this fix?

**Comment by ruihe774** [comment]
> Thanks, can you add a test that showed the issue before this fix?

Added

File: py-polars/tests/unit/io/test_csv.py [hunk 1]
```diff
@@ -2221,3 +2221,9 @@ def test_projection_applied_on_file_with_no_rows_16606(tmp_path: Path) -> None:
 
     out = pl.scan_csv(path).select(columns).collect().columns
     assert out == columns
+
+
+@pytest.mark.write_disk()
+def test_write_csv_to_dangling_file(df_no_lists: pl.DataFrame, tmp_path: Path) -> None:
```
Comment (ritchie46) [inline py-polars/tests/unit/io/test_csv.py:2227]:
```suggestion
def test_write_csv_to_dangling_file_17328(df_no_lists: pl.DataFrame, tmp_path: Path) -> None:
```

**Comment by ruihe774** [comment]
It is not necessary to extract `.buffer` from `TextIOWrapper` after #17315 - `TextIOWrapper` is also handled. We can simply remove these lines.

**Comment by ruihe774** [comment]
I'm sorry that there was a bug in #17315, namely the handling of `TextIOWrapper` in Windows and with non-UTF8 encoding. I've fixed it in 15ae286.

Now `TextIOWrapper` with non-UTF8 encoding is denied to avoid incorrect encoding. Previously `write_csv` does not check that.

**Review by ritchie46 — APPROVED** [review]

**Pull request merged**